python/src/opendp/_extrinsics/synth/__init__.py

```diff
@@ -0,0 +1,61 @@
+from opendp.mod import Domain, Metric
+import opendp.prelude as dp
+
+
+class Synthesizer:
+    def __init__(self):
+        self._is_fitted = False
+
+    def fit(self, data):
+        assert not self._is_fitted, "Synthesizer is already fitted"
+
+    def sample(self, num_records):
+        assert self._is_fitted, "Synthesizer must be fitted first"
+
+    def releasable(self):
+        assert self._is_fitted, "Synthesizer must be fitted first"
```

Comment:
Ditto previous comments about `assert` for input validation. (I really wish there wasn't that caveat on how it works!)